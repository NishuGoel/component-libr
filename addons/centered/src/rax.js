/** @jsx createElement */
// eslint-disable-next-line import/no-extraneous-dependencies
import { createElement } from 'rax';
// eslint-disable-next-line import/no-extraneous-dependencies
import View from 'rax-view';
import styles from './styles';

export default function(storyFn) {
  return (
    <View style={styles.style}>
      <View style={styles.innerStyle}>{storyFn()}</View>
    </View>
  );
}

if (module && module.hot && module.hot.decline) {
  module.hot.decline();
}
