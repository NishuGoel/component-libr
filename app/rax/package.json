{
  "name": "@storybook/rax",
  "version": "5.2.0-alpha.44",
  "description": "Storybook for Rax: Develop Rax Component in isolation.",
  "keywords": [
    "storybook",
    "rax"
  ],
  "homepage": "https://github.com/storybookjs/storybook/tree/master/app/rax",
  "bugs": {
    "url": "https://github.com/storybookjs/storybook/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/storybookjs/storybook.git",
    "directory": "app/rax"
  },
  "license": "MIT",
  "main": "dist/client/index.js",
  "jsnext:main": "src/client/index.js",
  "bin": {
    "build-storybook": "./bin/build.js",
    "start-storybook": "./bin/index.js",
    "storybook-server": "./bin/index.js"
  },
  "scripts": {
    "prepare": "node ../../scripts/prepare.js"
  },
  "dependencies": {
    "@storybook/core": "5.2.0-alpha.44",
    "babel-preset-rax": "^1.0.0-beta.0",
    "common-tags": "^1.8.0",
    "core-js": "^2.6.2",
    "global": "^4.3.2",
    "regenerator-runtime": "^0.12.1"
  },
  "devDependencies": {
    "rax": "^0.6.5"
  },
  "peerDependencies": {
    "babel-loader": "^7.0.0 || ^8.0.0",
    "rax": "^0.4.0 || ^1.0.0"
  },
  "publishConfig": {
    "access": "public"
  }
}
