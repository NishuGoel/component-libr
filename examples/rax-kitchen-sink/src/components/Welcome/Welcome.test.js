import { createElement } from 'rax';
import renderer from 'rax-test-renderer';
import Welcome from './index';

test('Welcome render correctly', () => {
  const tree = renderer.create(<Welcome />).toJSON();

  expect(tree).toMatchSnapshot();
});
