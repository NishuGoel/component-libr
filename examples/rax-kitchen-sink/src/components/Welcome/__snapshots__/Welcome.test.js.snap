// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome render correctly 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        "Welcome to storybook for Rax",
      ],
      "style": Object {
        "alignContent": "flex-start",
        "border": "0 solid black",
        "boxSizing": "border-box",
        "display": "block",
        "flexDirection": "column",
        "flexShrink": 0,
        "fontSize": 40,
        "fontWeight": "bold",
        "lineHeight": 50,
        "position": "relative",
        "whiteSpace": "pre-wrap",
      },
      "tagName": "SPAN",
    },
    Object {
      "attributes": Object {
        "fallbackSource": Object {
          "uri": "https://user-images.githubusercontent.com/677114/49848760-999e7d00-fe11-11e8-978f-264ea31f6739.png",
        },
        "resizeMode": "cover",
        "src": "./rax-logo.png",
      },
      "eventListeners": Object {
        "error": [Function],
        "load": [Function],
      },
      "style": Object {
        "display": "flex",
        "height": 100,
        "margin": "0 auto",
        "objectFit": "cover",
        "width": 100,
      },
      "tagName": "IMG",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            "This is a UI component dev environment for your Rax app.",
          ],
          "style": Object {
            "alignContent": "flex-start",
            "border": "0 solid black",
            "boxSizing": "border-box",
            "display": "block",
            "flexDirection": "column",
            "flexShrink": 0,
            "fontSize": 12,
            "position": "relative",
            "whiteSpace": "pre-wrap",
          },
          "tagName": "SPAN",
        },
      ],
      "style": Object {
        "alignContent": "flex-start",
        "border": "0 solid black",
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "flexShrink": 0,
        "marginBottom": 10,
        "position": "relative",
      },
      "tagName": "DIV",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            "We&#x27;ve added some basic stories inside the &#x27;src/stories&#x27; directory. A story is a single state of one or more UI components. You can have as many stories as you want. (Basically a story is like a visual test case.)",
          ],
          "style": Object {
            "alignContent": "flex-start",
            "border": "0 solid black",
            "boxSizing": "border-box",
            "display": "block",
            "flexDirection": "column",
            "flexShrink": 0,
            "fontSize": 12,
            "position": "relative",
            "whiteSpace": "pre-wrap",
          },
          "tagName": "SPAN",
        },
      ],
      "style": Object {
        "alignContent": "flex-start",
        "border": "0 solid black",
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "flexShrink": 0,
        "marginBottom": 10,
        "position": "relative",
      },
      "tagName": "DIV",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            "See these sample",
          ],
          "style": Object {
            "alignContent": "flex-start",
            "border": "0 solid black",
            "boxSizing": "border-box",
            "display": "block",
            "flexDirection": "column",
            "flexShrink": 0,
            "fontSize": 12,
            "position": "relative",
            "whiteSpace": "pre-wrap",
          },
          "tagName": "SPAN",
        },
      ],
      "style": Object {
        "alignContent": "flex-start",
        "border": "0 solid black",
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "flexShrink": 0,
        "marginBottom": 10,
        "position": "relative",
      },
      "tagName": "DIV",
    },
    Object {
      "attributes": Object {
        "role": "button",
      },
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                "Button",
              ],
              "style": Object {
                "alignContent": "flex-start",
                "border": "0 solid black",
                "boxSizing": "border-box",
                "display": "block",
                "flexDirection": "column",
                "flexShrink": 0,
                "fontSize": 12,
                "position": "relative",
                "whiteSpace": "pre-wrap",
              },
              "tagName": "SPAN",
            },
          ],
          "style": Object {
            "alignContent": "flex-start",
            "border": "0 solid black",
            "boxSizing": "border-box",
            "display": "flex",
            "flexDirection": "column",
            "flexShrink": 0,
            "marginBottom": 10,
            "position": "relative",
          },
          "tagName": "DIV",
        },
      ],
      "style": Object {
        "alignContent": "flex-start",
        "border": "0 solid black",
        "boxSizing": "border-box",
        "cursor": "pointer",
        "display": "flex",
        "flexDirection": "column",
        "flexShrink": 0,
        "position": "relative",
      },
      "tagName": "DIV",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            "for a component called &#x27;Button&#x27;.",
          ],
          "style": Object {
            "alignContent": "flex-start",
            "border": "0 solid black",
            "boxSizing": "border-box",
            "display": "block",
            "flexDirection": "column",
            "flexShrink": 0,
            "fontSize": 12,
            "position": "relative",
            "whiteSpace": "pre-wrap",
          },
          "tagName": "SPAN",
        },
      ],
      "style": Object {
        "alignContent": "flex-start",
        "border": "0 solid black",
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "flexShrink": 0,
        "marginBottom": 10,
        "position": "relative",
      },
      "tagName": "DIV",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            "Just like that, you can add your own components as stories. You can also edit those components and see changes right away. (Try editing the &#x27;Button&#x27; stories located at &#x27;src/stories/index.js&#x27; .)",
          ],
          "style": Object {
            "alignContent": "flex-start",
            "border": "0 solid black",
            "boxSizing": "border-box",
            "display": "block",
            "flexDirection": "column",
            "flexShrink": 0,
            "fontSize": 12,
            "position": "relative",
            "whiteSpace": "pre-wrap",
          },
          "tagName": "SPAN",
        },
      ],
      "style": Object {
        "alignContent": "flex-start",
        "border": "0 solid black",
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "flexShrink": 0,
        "marginBottom": 10,
        "position": "relative",
      },
      "tagName": "DIV",
    },
  ],
  "style": Object {
    "alignContent": "flex-start",
    "border": "0 solid black",
    "boxSizing": "border-box",
    "display": "flex",
    "flexDirection": "column",
    "flexShrink": 0,
    "fontFamily": "\\"Helvetica Neue\\", Helvetica, \\"Segoe UI\\", Arial, freesans, sans-serif",
    "margin": 15,
    "position": "relative",
  },
  "tagName": "DIV",
}
`;
